src/main/java/net/minecraftforge/fmp/microblock/MicroblockDelegate.java

```diff
@@ -0,0 +1,174 @@
+package net.minecraftforge.fmp.microblock;
+
+import com.google.common.base.Optional;
+
+import net.minecraft.block.Block;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.Multipart;
+
+/**
+ * Represents an object to which microblock events are delegated. This allows for custom interaction and data storage
+ * inside the microblock.
+ */
+public class MicroblockDelegate
+{
+    
+    protected final IMicroblock delegated;
+
+    public MicroblockDelegate(IMicroblock delegated)
+    {
+        this.delegated = delegated;
+    }
+
+    /**
+     * Harvests this microblock, removing it from the container it's in.
+     */
+    public boolean harvest(EntityPlayer player, RayTraceResult hit)
+    {
+        return false;
+    }
+
+    /**
+     * Gets the strength of a player against this microblock. Not to be confused with
+     * {@link Multipart#getHardness(RayTraceResult)}. This also takes potion effects and tools into account.
+     */
+    public Optional<Float> getStrength(EntityPlayer player, RayTraceResult hit)
+    {
+        return Optional.absent();
+    }
+
+    /**
+     * Called when a part in the same block block changes.
+     */
+    public void onPartChanged(IMultipart part)
+    {
+    }
+
+    /**
+     * Called when a neighbor block changes.
+     */
+    public void onNeighborBlockChange(Block block)
+    {
+    }
+
+    /**
+     * Called when a neighbor tile changes.
+     */
+    public void onNeighborTileChange(EnumFacing facing)
+    {
+    }
+
+    /**
+     * Called when this microblock is added to the container.
+     */
+    public void onAdded()
+    {
+    }
+
+    /**
+     * Called right before this microblock is removed from the container.
+     */
+    public void onRemoved()
+    {
+    }
+
+    /**
+     * Called when the chunk this microblock is in is loaded.
+     */
+    public void onLoaded()
+    {
+    }
+
+    /**
+     * Called when the chunk this microblock is in is unloaded.
+     */
+    public void onUnloaded()
+    {
+    }
+
+    /**
+     * Called when a player right-clicks this microblock. Return true to play the right-click animation.
+     */
+    public Optional<Boolean> onActivated(EntityPlayer player, EnumHand hand, ItemStack heldItem, RayTraceResult hit)
```

Comment:
true = animate arm
false = click failed
null = pass through to the microblock itself, then to whatever is needed

When this was added EnumActionResult wasn't a thing yet, but I'll switch over to it, yes :)